sim/pokemon.js

```diff
@@ -35,7 +35,7 @@ class Pokemon {
 		let pokemonScripts = this.battle.data.Scripts.pokemon;
 		if (pokemonScripts) Object.assign(this, pokemonScripts);
 
-		if (typeof set === 'string') set = {name: set};
+		if (typeof set === 'string') set = new this.battle.Data.PokemonSet({name: set});
```

Comment:
???

I don't think `PokemonSet` needs to be a class; it's fine as an interface...

Reply:
I needed to construct a `PokemonSet` object in order to properly TS `fastUnpackTeam`, and I figured it would make sense to make a constructor, which let me add a bunch of documentation to all the properties. Looking at it now, I can see a way to accomplish the same thing with a few instances of `//@ts-ignore` instead of making a class, but then we'd lose all that nice documentation I spent at least an hour writing.

Reply:
Can you point me to which line needs `ts-ignore` if PokemonSet were an interface?

Reply:
I'd probably still do `/** @type {PokemonSet} */ ({})` rather than `new PokemonSet`, though.

Reply:
I didn't think that kind of thing worked. Is that a new feature I missed?

Reply:
It's just how you cast in `.js`-TypeScript.

Reply:
It's the equivalent of `<PokemonSet>{}` or `{} as PokemonSet` in real TypeScript.